features/cloud.py

```diff
@@ -0,0 +1,373 @@
+import os
+import logging
+import pycloudlib  # type: ignore
+import time
+import yaml
+
+from typing import Optional
+
+
+class Cloud:
+    """Base class for cloud providers that should be tested through behave.
+
+    :param tag:
+        A tag to be used when creating the resources on the cloud provider
+    """
+
+    name = ""
+
+    def __init__(self, tag):
```

Comment:
For any method and function we touch, let's try to add typing hints to better describe params and returns (and validated with `tox -e mypy`)

```suggestion
    def __init__(self, tag: str)  -> None:
```